src/Orchard/Mvc/ViewEngines/ThemeAwareness/ThemeAwareViewEngine.cs

```diff
@@ -93,12 +94,17 @@ public class ThemeAwareViewEngine : IThemeAwareViewEngine {
                     .Reverse()  // reverse from (C <= B <= A) to (A => B => C)
                     .Where(fd => DefaultExtensionTypes.IsModule(fd.Extension.ExtensionType));
 
-                var allLocations = enabledModules.Concat(enabledModules)
-                    .Select(fd => fd.Extension.Location + "/" + fd.Extension.Id)
-                    .Distinct(StringComparer.OrdinalIgnoreCase)
+                var moduleVirtualPaths = enabledModules
+                    .Select(fd => fd.Extension.VirtualPath)
+                    .Distinct(StringComparer.OrdinalIgnoreCase) // is Distinct guaranty to keep order?
                     .ToList();
 
-                var moduleParams = new CreateModulesViewEngineParams { VirtualPaths = allLocations };
+                var moduleLocations = enabledModules

```

Comment:
Seems like the original code had a minor bug: enabledModules is duplicated by concatination then filtered by distinct.. other than that all seems to be OK.